### DISCUSSION TITLE
Could ccache get an option to cache into memory?

### DISCUSSION THREAD

**Discussion by camelcigarettes** [ccache/ccache#1399] [Development]
i think it does it by disk space, it would go well with *nix because its scalable and ram is cheap, i myself have 128gb of ram on my hp envy gaming pc

all you need is ram drive libraries and other stuff, think you can just copy the files over to the drive when needed, havent really looked at the source code of ccache

**Comment by jrosdahl** [comment]
It's already possible to have the cache in memory by putting the ccache directory on a RAM disk.

Example on Linux:

```
export CCACHE_DIR=/dev/shm/my-ccache-in-memory
```

Another possibility is to use the Redis remote storage backend.

**Comment by camelcigarettes** [comment]
Ty! I needed something for speedy compiles on my hp envy with 128gb of ram

**Reply by afbjorklund** [reply]
Make sure to benchmark, since the kernel is quite good at caching files for read - your biggest benefit is when **writing** files

If you use Redis for a persistent cache, then make sure to run enough processes or use a multi-threaded variant like KeyDB*

And please do post your findings, if it makes a difference for you

\* see https://github.com/ccache/ccache/wiki/Redis-storage

**Reply by afbjorklund** [reply]
I made some experiments with [single-file formats](https://github.com/ccache/ccache/issues/1118), for more efficient storage and then loading into memory for access...

It wasn't included in the end, but there was both archive (.tar) and database (.mdb) formats. Now just using dir or redis.

They could be added back as storage plugins.

* https://github.com/ccache/ccache/issues/1214
